dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -960,33 +959,12 @@ protected Contentlet find(final String inode) throws ElasticsearchException, Dot
 
 	@Override
 	protected Contentlet findContentletByIdentifier(String identifier, Boolean live, Long languageId) throws DotDataException {
-		try {
-			StringWriter query= new StringWriter();
-			query.append(" +identifier:" + identifier);
-			query.append(" +languageid:" + languageId);
-			query.append(" +deleted:false");
-
-            final SearchSourceBuilder searchSourceBuilder = createSearchSourceBuilder(query.toString());
-			IndiciesInfo info=APILocator.getIndiciesAPI().loadIndicies();
-
-			final  SearchRequest searchRequest = new SearchRequest();
-			searchRequest.source(searchSourceBuilder);
-			searchRequest.indices((live ? info.getLive() : info.getWorking()));
-
-            final SearchResponse response = Sneaky.sneak(()->
-                    RestHighLevelClientProvider.getInstance().getClient().search(searchRequest, RequestOptions.DEFAULT));
-            SearchHits hits = response.getHits();
-
-			return find(hits.getAt(0).getSourceAsMap().get("inode").toString());
-		}
-		// if we don't have the con in this language
-		catch(ArrayIndexOutOfBoundsException aibex){
-			return null;
-		}
-		catch (Exception e) {
-			throw new ElasticsearchException(e.getMessage(), e);
-
-		}
+        final ContentletVersionInfo cvi = APILocator.getVersionableAPI().getContentletVersionInfo(identifier, languageId);
```

Comment:
An IT should be implemented for this change in case a test for the old logic didn't exist